iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -528,6 +528,45 @@ private async Task OpenMethodsReceiverLinkAsync(AmqpIoTSession amqpIoTSession, s
             }
         }
 
+        public async Task DisableTwinPatchAsync(TimeSpan timeout)
```

Comment:
presumably the timeout is for the entire operation, but we're using it in every method call below, which would mean the total timeout would be multiples of `timeout`, right?

Seems we should be decrementing timeout as we go.

Reply:
Timeout is for the underlying amqp operations. We just need to pass it down and the amqp library takes care of rest.